File: customer-java-webflux/src/main/java/com/prez/lib/tracing/CleanLoggingContextFilter.java
package com.prez.lib.tracing;

import org.slf4j.MDC;
import org.springframework.web.server.ServerWebExchange;
import org.springframework.web.server.WebFilter;
import org.springframework.web.server.WebFilterChain;
import reactor.core.publisher.Mono;

public class CleanLoggingContextFilter implements WebFilter {
  @Override
  public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
    return chain.filter(exchange)
        .doOnTerminate(MDC::clear);
  }
}

File: customer-kotlin-coroutines-controllers/src/main/kotlin/com/prez/lib/tracing/CleanLoggingContextFilter.kt
package com.prez.lib.tracing

import org.slf4j.MDC
import org.springframework.web.server.ServerWebExchange
import org.springframework.web.server.WebFilter
import org.springframework.web.server.WebFilterChain
import reactor.core.publisher.Mono

class CleanLoggingContextFilter : WebFilter {
    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
        return chain.filter(exchange).doOnTerminate(MDC::clear)
    }
}
